core/src/smartcontracts/isi/permissions/checks.rs

```diff
@@ -45,15 +33,13 @@ pub fn check_instruction_permissions(
 fn check_permissions_directly(
     account_id: &AccountId,
     instructions: &[Instruction],
-    is_instruction_allowed: &IsInstructionAllowedBoxed,
+    is_instruction_allowed: &dyn Judge<Operation = Instruction>,
```

Comment:
Quick question,  have you considered an opaque type? I don't think the distinction matters right now, but in the future it might contrast with how we distinguish between ISI and WASM. Static dispatch could work here just as well.

Reply:
Do you mean something like enum or to introduce generics in this function?

Reply:
`impl Judge<Operation = Instruction>`. Dynamic dispatch is fine, though.